upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go

```diff
@@ -519,6 +529,17 @@ func (v *AutoscalingGroup) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *Autos
 			changes.Subnets = nil
 		}
 
+		if changes.MaxInstanceLifetime != nil {
+			request.MaxInstanceLifetime = e.MaxInstanceLifetime
+			changes.MaxInstanceLifetime = nil
+		} else {
+			//If max instance lifetime has been removed from spec and we have a max instance lifetime
+			//on the "physical" ASG, we need to reset the value to 0
+			if a.MaxInstanceLifetime != nil {
```

Comment:
I don't think `a.MaxInstanceLifetime` can ever be nil. If we unset `MaxInstanceLifetime` in the spec, this value would be `0`, otherwise it would be `> 0`. In other cases, `changes.MaxInstanceLifetime` would be `nil` and we'd never enter this block.

Reply:
Fixed in 5073d5b